components/eventing-controller/pkg/backend/nats/jetstream/jetstream.go

```diff
@@ -177,13 +177,53 @@ func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscriptio
 		return err
 	}
 
+	if err := js.checkSubscriptionConfig(subscription, log); err != nil {
+		return err
+	}
+
 	if err := js.checkNATSSubscriptionsCount(subscription); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// CheckSubscriptionConfig checks that the latest Subscription Config changes are propagated to the consumer.
+// In our case config contains only the "maxInFlightMessages" property, which is the maxAckPending on the consumer side.
+func (js *JetStream) checkSubscriptionConfig(subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
+	for _, subject := range subscription.Status.CleanEventTypes {
```

Comment:
I am really not sure whether we should use the status value also here. We can discuss this with others what the best practice is.

Reply:
this is the way how we usually do it. The cleanEventTypes map is always populated before this function call, so I think that's ok. It will change in the v2 of the subscription crd.